src/main/java/com/mitchej123/hodgepodge/mixins/late/thaumcraft/MixinBlockMagicalLeaves.java

```diff
@@ -0,0 +1,124 @@
+package com.mitchej123.hodgepodge.mixins.late.thaumcraft;
+
+import java.util.Random;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.material.Material;
+import net.minecraft.world.World;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+
+import thaumcraft.common.blocks.BlockMagicalLeaves;
+
+@Mixin(value = BlockMagicalLeaves.class, remap = false)
+public abstract class MixinBlockMagicalLeaves extends Block {
+
+    @Shadow
+    int[] adjacentTreeBlocks;
+
+    @Shadow
+    protected abstract void removeLeaves(World par1World, int par2, int par3, int par4);
+
+    private MixinBlockMagicalLeaves(Material material) {
+        super(material);
+    }
+
+    /**
+     * @author HoleFish
+     * @reason Fix frequent ticking caused by old method
+     */
+    @Overwrite
+    public void updateTick(World worldIn, int x, int y, int z, Random random) {
```

Comment:
Does this method essentially just replace the worldIn.setBlock(...) in thaum with worldIn.setBlockMetadataWithNotify(...) or am I missing something?

Reply:
Yes. Ive tried redirection but it surprisingly didnt work. Maybe I should try harder;)

Reply:
Something like this should do the trick
```
@Redirect(
            method = "updateTick",
            at = @At(
                    value = "INVOKE",
                    target = "Lnet/minecraft/world/World;setBlock(IIILnet/minecraft/block/Block;II)Z",
                    remap = false))
    public boolean hodgepode$xxx(World world, int x, int y, int z, Block block, int meta, int flag) {}
```

Reply:
I tried exactly what you said 2 days ago and it did't work. Now I try again and it works well. Maybe my codes need a nap :trollface: